pkg/vsphere/tasks/waiter.go

```diff
@@ -70,10 +75,18 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 		}
 
 		log.Errorf("task failed: %s", err)
-		if !isTaskInProgress(err) {
+		if !needsRetry(target, err) {
```

Comment:
I feel it's more clear to have one single retry detection method. And the needsFix detection is easy to do.